inference_server/launcher/gputranslator.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2025 The llm-d Authors.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# 	http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+GPU Translator
+"""
+
+from typing import Dict
+
+import pynvml
+
+
+# VLLM process manager
+class GpuTranslator:
+
+    def __init__(self):
+        """
+        Initialize GPU Translator
+        """
+        self.device_count = 0
+        self.mapping = {}
+        # Create reverse mapping
+        self.reverse_mapping = {}
+
+    def _populate_mapping(self):
+        """
+        Creates mapping and reverse_mapping for the GPU Translator
+        """
+        pynvml.nvmlInit()
+        self.device_count = pynvml.nvmlDeviceGetCount()
+        for index in range(self.device_count):
+            handle = pynvml.nvmlDeviceGetHandleByIndex(index)
+            uuid = pynvml.nvmlDeviceGetUUID(handle).decode("utf-8")
```

Comment:
Don't you want to handle NVMLError, as outlined in https://pypi.org/project/nvidia-ml-py/ ?